src/pages/Portfolio/ImpactDashboard/ImpactDashboardPage.vue

```diff
@@ -115,7 +122,14 @@ export default {
 		this.showTeamChallenge = teamsChallengeEnable && this.allowedTeams.length > 0;
 
 		if (this.$route?.query?.goal_saved) {
-			this.$showTipMsg('Goal saved');
+			const badgeName = this.route?.query?.goal_saved ?? '';
+			const userPreferences = portfolioQueryData.my?.userPreferences ?? null;
+
+			this.storeGoal({ userPreferences, badgeName }).then(() => {
+				this.$showTipMsg('Goal saved');
```

Comment:
Should this logic be happening on `mounted`? Due to the tooltip not being needed server-side.